Civi/Api4/Service/Spec/Provider/AddressCreationSpecProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Civi\Api4\Service\Spec\Provider;
+
+use Civi\Api4\Service\Spec\FieldSpec;
+use Civi\Api4\Service\Spec\RequestSpec;
+
+
+class AddressCreationSpecProvider implements SpecProviderInterface {
+
+  /**
+   * @param RequestSpec $spec
+   */
+  public function modifySpec(RequestSpec $spec) {
+    $action = $spec->getAction();
+    $spec->getFieldByName('contact_id')->setRequired(TRUE);
+    $spec->getFieldByName('location_type_id')->setRequired(TRUE);
+
+    $extraFields = [
+      'world_region' => [
+        'type' => 'String',
+        'title' => ts('World Region'),
+      ],
+    ];
+    if ($action !== 'get') {
+      $extraFields = array_merge($extraFields, [
+        'street_parsing' => [
+          'type' => 'Boolean',
+          'title' => ts('Street Parsing'),
+          'description' => ts('Optional param to indicate you want the street_address field parsed into individual params'),
+        ],
+        'skip_geocode' => [
+          'type' => 'Boolean',
+          'title' => ts('Skip geocode'),
+          'description' => ts('Optional param to indicate you want to skip geocoding (useful when importing a lot of addresses at once, the job \'Geocode and Parse Addresses\' can execute this task after the import)'),
+        ],
+        'fix_address' => [
+          'type' => 'Boolean',
+          'title' => ts('Fix address'),
+          'description' => ts('When true, apply various fixes to the address before insert. Default true.'),
+          'default' => TRUE,
+        ],
+      ]);
```

Comment:
I'd say these are parameters and not fields. A little background:

- In api3 fields & params are mismashed into a single array, which is confusing and problematic.
- In api4 they are separate. The create action has a `$values` param, which takes a list of field values. It can also have other params like `$checkPermissions`. You can add new ones like so:

1. Override the `Create` method for an entity.
2. Declare a `protected` class variable. It needs to have a default value and an annotation declaring its data type (string, int, bool, etc).
3. That's it, it will appear in the api explorer.

Reply:
FYI @eileenmcnaughton ^
This is basically doing away with the api3-style `$params['options']` array in favor of each "option" being its own param.

Reply:
@colemanw I created a new file ```Civi/Api4/Action/Address/Create.php``` and here is it's content:
```php
namespace Civi\Api4\Action\Address;

use Civi\Api4\Generic\Result;
use Civi\Api4\Action\Create as DefaultCreate;

/**
 * @inheritDoc
 */
class Create extends DefaultCreate {

  /**
   * @var boolean
   */
  protected $street_parsing = TRUE;

}
```
but wasn't able to get this new paramter in API explorer. Is there something else I need to do?

Reply:
Try `@var bool` instead of `@var boolean` I'm not sure why this makes a difference, but it does.

Also, convention in Api4 is to use camelCase for param names.